src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -807,13 +807,33 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 	 * and clears it if it isn't.
 	 * @param connection The connection to fill in or update
 	 */
-	private async fillInOrClearAzureToken(connection: interfaces.IConnectionProfile): Promise<boolean> {
-		if (connection.authenticationType !== Constants.azureMFA && connection.authenticationType !== Constants.azureMFAAndUser) {
+	private async fillInOrClearToken(connection: interfaces.IConnectionProfile): Promise<boolean> {
+		if (connection.authenticationType !== Constants.azureMFA
+			&& connection.authenticationType !== Constants.azureMFAAndUser
+			&& connection.authenticationType !== Constants.dstsAuth) {
 			connection.options['azureAccountToken'] = undefined;
 			return true;
 		}
+
 		let azureResource = this.getAzureResourceForConnection(connection);
 		const accounts = await this._accountManagementService.getAccounts();
+
+		if (connection.authenticationType === Constants.dstsAuth) {
+			let dstsAccounts = accounts.filter(a => a.key.providerId.startsWith('dstsAuth'));
+			if (dstsAccounts.length <= 0) {
+				connection.options['azureAccountToken'] = undefined;
+				return false;
+			}
+
+			dstsAccounts[0].key.providerArgs = {
+				serverName: connection.serverName,
+				databaseName: connection.databaseName
+			};
+
+			connection.options['azureAccountToken'] = await (await this._accountManagementService.getAccountSecurityToken(dstsAccounts[0], '', null)).token;
```

Comment:
nit: we usually use `undefined` instead of `null` unless we explicitly need null.  These are different types so it's helpful to be consistent throughout our codebase.